### PR TITLE
pass along version to httpd/handler

### PR DESCRIPTION
**Pull request by neonstalwart**
i'm not sure if this is *how* you want to do this.  if there's a way which seems better, let me know and i'll change this.

### CODE REVIEW

**Comment by otoolep** [comment]
Works for me +1. @corylanou?

@neonstalwart -- can you show an example response from the server, with this change in place?

**Comment by neonstalwart** [comment]
I'm on my way out of town but I might be able to do something by Monday.

File: services/httpd/handler.go [hunk 1]
```diff
@@ -134,7 +133,7 @@ func (h *Handler) SetRoutes(routes []route) {
 		if r.gzipped {
 			handler = gzipFilter(handler)
 		}
-		handler = versionHeader(handler, h.version)
+		handler = versionHeader(handler, h)
```
Comment (corylanou) [inline services/httpd/handler.go]: I typically prefer to only send what is needed to the receiving method, which in this case is just the version string.  It makes the receiving function easy to understand what is actually needed.

File: services/httpd/handler.go [hunk 2]
```diff
@@ -662,9 +661,9 @@ func gzipFilter(inner http.Handler) http.Handler {
 
 // versionHeader taks a HTTP handler and returns a HTTP handler
 // and adds the X-INFLUXBD-VERSION header to outgoing responses.
-func versionHeader(inner http.Handler, version string) http.Handler {
+func versionHeader(inner http.Handler, h *Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.Header().Add("X-InfluxDB-Version", version)
+		w.Header().Add("X-InfluxDB-Version", h.Version)
```
Comment (corylanou) [inline services/httpd/handler.go]: Related to the above comment, same thing.  Restricting scope for arguments is important.

Comment (corylanou) [inline services/httpd/handler.go]: oh, I see what you did now that I looked at the file as a whole.  I guess I'm fine making the signatures consistent.  Ignore my previous comments :-)

**Comment by corylanou** [comment]
This looks good.  I think we used to have a test that verified the version.  I think it got lost in the shuffle.  It would be nice to add that back in and make sure it works.

Once again, thanks for the work @neonstalwart!

+1

File: services/httpd/handler.go (hunk 2 above)
Comment (neonstalwart) [inline services/httpd/handler.go]: actually the reason for it is that at the time `versionHeader` is called `h.Version` has not been set. so I passed a pointer to `h` and by the time the `HandlerFunc` is called, `h.Version` has been set. there were alternatives to this but I found this one to be my preference.

**Comment by otoolep** [comment]
Unfortunately this change does not work. `s.Version` has not yet been set when the handler's version field is set.

I am looking into a fix.

**Comment by otoolep** [comment]
Fixed by https://github.com/influxdb/influxdb/pull/2969

**Pull request merged**